### PR TITLE
Add newline for pund-keywords in `IfConfigDeclSyntax`

### PR DESCRIPTION
**Pull request by kimdv**

### CODE REVIEW

File: Tests/SwiftSyntaxBuilderTest/IfConfigDeclSyntaxTests.swift [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        )
+      },
+      poundEndif: .poundEndifKeyword(leadingTrivia: .newline)
+    )
+
+    AssertBuildResult(
+      buildable,
+      """
+
+      #if DEBUG
```
Comment (kimdv) [inline Tests/SwiftSyntaxBuilderTest/IfConfigDeclSyntaxTests.swift:58]: The only "downside" is that it also counts for the `#if` and not only `#elseIf`

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by kimdv** [comment]
@swift-ci please test

File: Tests/SwiftSyntaxBuilderTest/IfConfigDeclSyntaxTests.swift (hunk 1 above)
Comment (ahoppen) [reply Tests/SwiftSyntaxBuilderTest/IfConfigDeclSyntaxTests.swift:58]: This should probably be fixed if you add a check to `shouldOmitNewline` that returns `true` if to token is the first in the source file, i.e. if `node.previousToken(view: .sourceAccurate) == nil`

**Comment by kimdv** [comment]
@swift-ci please test

**Review by ahoppen — APPROVED** [review]

**Pull request merged**